cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -212,67 +216,51 @@ def migrate_from_modulestore(
     status.increment_completed_steps()
 
     status.set_state(MigrationStep.IMPORTING_STRUCTURE.value)
+
+    # @@TODO: I propose to make it better:
+    # Implement as a MigrationContext dataclass
+    # encapsulate in it the following migration artibutes
+    # Context:
+        # existing_source_to_target_keys:
+        # content_by_filename:
+        # target_package_id:
+        # replace_existing:
+        # preserve_url_slugs:
+    # This will reduce parameter count from 10+ to 2-3
```

Comment:
As long as we're talking about a `frozen=True` dataclass, that sounds good to me @cmltaWt0 .

Reply:
Implemented.